### ISSUE TITLE
Add a wrapper

### ISSUE THREAD

**Issue by ericsagnes** [styx-static/styx#2]
imported from https://github.com/NixOS/nixos-weekly/issues/23#issuecomment-251165661

The next level of power would be to wrap it so that it's not necessary to ship styx. For example if it's made part of nixpkgs. Then just need a top-level default.nix and the content of the site are needed. The default.nix would look like:

``` nix
with import <nixpkgs> {};
stdenv.mkStyxSite {
  name = "my-site-name";
  conf = import ./conf.nix; # or keep it inline
  src = ./.;
  # ...
}
```

**Comment by ericsagnes** [comment]
I have played a little with this idea, and I am not sure if this can be implemented.

This require to call `nix-build` (`styx build` is just a wrapper around nix-build) during a nix build (`mkStyxSite`), and Nix dislike it.

It throws the following error:

```
error: creating directory ‘/nix/var’: Permission denied
```

But maybe I am just taking a wrong approach?

**Comment by ericsagnes** [comment]
Took a different approach, so now styx sites can be builded with a simple `callPackage` call.
Example integration with [NixOps](https://github.com/styx-static/styx/blob/1aa35cdde51b890259eeeb621b19ba22410460b7/src/doc/deployment.adoc#nixops).

Note: This require changes in nixpkgs that are not yet commited upstream, but that will be for the 0.4.0 release.

Closed via 163281e0c25d88e5bb3cd91889fa3b8028b98ccb.
